File: FitToFit/FitToFit.Model/Novosti.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FitToFit.Model
{
    public class Novosti
    {
        public int NovostId { get; set; }

        public string Naslov { get; set; } = null!;

        public string? Sadrzaj { get; set; }

        public bool? IsLiked { get; set; }

        public int BrojLajkova { get; set; }

        public DateTime DatumObjave { get; set; }

        public int KorisnikId { get; set; }

        public int? VrstaTreningaId { get; set; }

        public bool? IsRead { get; set; }
    }
}

File: FitToFit/UI/fittofit_mobile/lib/models/novosti.dart
import 'package:json_annotation/json_annotation.dart';

part 'novosti.g.dart';

@JsonSerializable()
// ignore: invalid_annotation_target
@JsonKey(name: 'isLiked')
class Novosti {
  int novostId;
  String naslov;
  String? sadrzaj;
  bool? isLiked;
  int brojLajkova;
  DateTime datumObjave;
  int korisnikId;
  int? vrstaTreningaId;

  Novosti(
      {required this.novostId,
      required this.naslov,
      this.sadrzaj,
      this.isLiked,
      required this.brojLajkova,
      required this.datumObjave,
      required this.korisnikId,
      this.vrstaTreningaId});

  factory Novosti.fromJson(Map<String, dynamic> json) =>
      _$NovostiFromJson(json);

  Map<String, dynamic> toJson() => _$NovostiToJson(this);
}
